### PR TITLE
Adding release process

### PR DESCRIPTION
**Pull request by jeremyf**
## Updating CHANGELOG.md format

f345e0398b3e1a8a57bd25157abd7c902268da68

I started exploring [github_changelog_generator][1] after learning of
it from =bess. This update pairs with a future commit in which I plan
to write the release procedures, which uses
[github_changelog_generator][1].

~enhancement

[1]:https://rubygems.org/gems/github_changelog_generator

## Adding release procedures to huborg

318eb1c64f1e17086e4f51763d61cea7b553d377

In a community based project, there are many moving parts. I wanted to
be clear about the steps used in releasing Huborg. The grant/revoke of
gem authority is something that can get sticky if you were commit a
bumped version and run `rake release`. You'll have tagged a release,
pushed to github, but not rubygems.

~enhancement

### CODE REVIEW

**Comment by jeremyf** [comment]
Note, the commit messages include `~enhancement`, I was hoping this would add a label of "enhancement" to the pull request. That did not appear to work.

File: Rakefile [hunk 1]
```diff
@@ -42,3 +42,21 @@ namespace :test do
...
+    ENV.fetch("CHANGELOG_GITHUB_TOKEN")
+  rescue KeyError
+    $stderr.puts %(To run `rake changelog` you need to have a CHANGELOG_GITHUB_TOKEN)
+    $stderr.puts %(set in ENV. (`export CHANGELOG_GITHUB_TOKEN="«your-40-digit-github-token»"`))
+    exit!(1)
+  end
+  config.user = 'samvera-labs'
+  config.project = 'huborg'
+  config.since_tag = 'v0.1.0' # The changes before v0.1.0 were not as helpful
+  config.future_release = "v#{ENV.fetch("FUTURE_RELEASE", Huborg::VERSION)"
```
Comment (jrgriffiniii) [inline Rakefile]: This is missing a trailing `}` for `v#{ENV.fetch(...)`

**Comment by jeremyf** [comment]
@jrgriffiniii Great catch. That was something I was holding off on running until I had this merged. I amended the previous commit and force pushed content.

**Comment by jrgriffiniii** [comment]
I need to rebase for the `README`, otherwise I will merge this.

**Pull request merged**